app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/dataSetDetail/DataSetDetailPresenter.kt

```diff
@@ -0,0 +1,72 @@
+package org.dhis2.usescases.datasets.dataSetTable.dataSetDetail
+
+import io.reactivex.Flowable
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.functions.BiFunction
+import io.reactivex.processors.PublishProcessor
+import org.dhis2.data.schedulers.SchedulerProvider
+import org.dhis2.usescases.datasets.dataSetTable.DataSetTableRepository
+import org.hisp.dhis.android.core.dataset.DataSetInstance
+import org.hisp.dhis.android.core.period.Period
+import timber.log.Timber
+
+class DataSetDetailPresenter(
+    val view: DataSetDetailView,
+    val repository: DataSetTableRepository,
+    val schedulers: SchedulerProvider
+) {
+    private val disposable = CompositeDisposable()
```

Comment:
I think you could add unit test for this :+1:

Reply:
Done!